### PR TITLE
feat: notification list component

### PR DESCRIPTION
**Pull request by rmathis**
Initial pass at the notification list component.

### CODE REVIEW

File: @uportal/notification-list/src/components/NotificationItem.vue [hunk 1]
```diff
@@ -0,0 +1,236 @@
...
+import DropdownHeader from 'bootstrap-vue/es/components/dropdown/dropdown-header';
+import DropdownItemButton from 'bootstrap-vue/es/components/dropdown/dropdown-item-button';
+import { library } from '@fortawesome/fontawesome-svg-core';
+import { faEllipsisV } from '@fortawesome/free-solid-svg-icons/faEllipsisV';
+import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome';
+import { format } from 'date-fns';
+
+library.add(faEllipsisV);
+
+/*eslint-disable */
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationItem.vue]: Which eslint rules were causing issues?

File: @uportal/notification-list/src/components/NotificationItem.vue [hunk 2]
```diff
@@ -0,0 +1,236 @@
...
+            const separator = categories && dueDate ? ' - ' : '';
+            return `${categories || ''}${separator}${dueDate || ''}`;
+        },
+        itemCategories() {
+            return this.notification.attributes.category
+                ? this.notification.attributes.category.join(', ')
+                : '';
+        },
+        itemDueDate() {
+            let dueDate = this.notification.dueDate;
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationItem.vue]: :art: could this be made `const`?

File: @uportal/notification-list/src/components/NotificationItem.vue [hunk 3]
```diff
@@ -0,0 +1,236 @@
...
+            if (!dueDate) {
+                return null;
+            }
+            let date = format(new Date(dueDate.time), 'MM/DD/YYYY');
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationItem.vue]: :art: could this be made `const`?

File: @uportal/notification-list/src/components/NotificationItem.vue (hunk 1 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationItem.vue]: nope, was a holdover from trying to fix the optional property chaining. removed

File: @uportal/notification-list/src/components/NotificationItem.vue (hunk 2 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationItem.vue]: done

File: @uportal/notification-list/src/components/NotificationItem.vue (hunk 3 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationItem.vue]: done

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 4]
```diff
@@ -115,10 +110,20 @@ export default {
...
             } catch (err) {
-                console.error(err);
+                this.err = err;
+            } finally {
+                this.loaded = true;
             }
         }
     },
+    mounted() {
+        console.log('mounted');
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]: left over log?

File: @uportal/notification-list/src/components/NotificationItem.vue [hunk 5]
```diff
@@ -0,0 +1,235 @@
...
+    }
+
+    .read {
+        background-color: #fff;
+        background-color: var(--notif-read-bg-color, #fff) !important;
+    }
+
+    .highlight {
+        background-color: honeydew;
+        background-color: var(--notif-highlight-bg-color, honeydew) !important;
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationItem.vue]: Are the `!important`s needed?

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 6]
```diff
@@ -1,21 +1,164 @@
...
+            default: () => ({
+                Announcement: '#6649bb',
+                Holds: '#487df9',
+                'To-Do': '#c85a89'
+            })
+        }
+    },
+    methods: {
+        gotoAction(action) {
+            let form = document.createElement('form');
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]: :art: could this be made `const`?

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 7]
```diff
@@ -1,21 +1,164 @@
...
     },
-    computed: {}
+    mounted() {
+        this.fetchNotificationData();
+
+        const queryParams = queryString.parse(location.search);
+        this.highlight = queryParams.highlight;
+    },
+    filters: {
+        dueDate: function(value) {
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]: :art: 

```suggestion
        dueDate (value) {
```

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 8]
```diff
@@ -1,21 +1,164 @@
...
+            if (!value) {
+                return '';
+            }
+            let date = format(new Date(value.time), 'MM/DD/YYYY');
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]:
```suggestion
            const date = format(new Date(value.time), 'MM/DD/YYYY');
```

File: @uportal/notification-list/src/components/NotificationList.vue [hunk 9]
```diff
@@ -1,21 +1,164 @@
...
     </div>
 </template>
 <script>
-// import oidc from '@uportal/open-id-connect';
+import oidc from '@uportal/open-id-connect';
+
+import NotificationItem from './NotificationItem';
+import queryString from 'query-string';
+
+/*eslint-disable */
```
Comment (ChristianMurphy) [inline @uportal/notification-list/src/components/NotificationList.vue]: What ESLint rules are being disabled?

Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: removed

File: @uportal/notification-list/src/components/NotificationList.vue (hunk 8 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: updated to const

File: @uportal/notification-list/src/components/NotificationList.vue (hunk 7 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: function is unnecessary in this component now, removed

File: @uportal/notification-list/src/components/NotificationList.vue (hunk 6 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationList.vue]: updated to const

File: @uportal/notification-list/src/components/NotificationItem.vue (hunk 5 above)
Comment (rmathis) [reply @uportal/notification-list/src/components/NotificationItem.vue]: no, removed

**Comment by ChristianMurphy** [comment]
@rmathis LGTM :ballot_box_with_check: :+1: 
This PR is still labeled `wip`, what else is needed for this to be merged?

File: @uportal/notification-list/README.md [hunk 10]
```diff
@@ -69,6 +77,28 @@ Url to connect to oidc. The default for this value is `/uPortal/api/v5-1/userinf
 <notification-list oidc-url="/uPortal/api/v5-1/userinfo"></notification-list>
 ```
 
+### `color-map`
+
+A map of colors associated with notification categories. The component will choose the first category defined to determine this.
+
+```html
+<notification-list
+    oidc-url='{
```
Comment (drewwills) [inline @uportal/notification-list/README.md]: `oidc-url`?

**Pull request merged**